lib/awful/key.lua

```diff
@@ -333,14 +337,36 @@ key.keygroups = {
         {"Right" , "Right" },
         {"Up"    , "Up"    },
         {"Down"  , "Down"  },
-    }
+    },
+    fkeys = {},
+    numpad = {
+        {"#87"   , "1"},
+        {"#88"   , "2"},
+        {"#89"   , "3"},
+        {"#83"   , "4"},
+        {"#84"   , "5"},
+        {"#85"   , "6"},
+        {"#79"   , "7"},
+        {"#80"   , "8"},
+        {"#81"   , "9"},
+        {"#90"   , "0"},
+        {"#91"   , "."},
+        {"#106"  , "/"},
+        {"#63"   , "*"},
+        {"#82"   , "-"},
+        {"#86"   , "+"},
+        {"#104"  , "\n"},
+    },
 }
 
 -- Technically, this isn't very popular, but we have been doing this for 12
 -- years and nobody complained too loudly.
 for i = 1, 10 do
     table.insert(key.keygroups.numrow, {"#" .. i + 9, i == 10 and 0 or i})
 end
+for i = 1, 12 do
```

Comment:
F keys goes beyond F12. I am not sure I remember, I think its F24. It needs to be double checked. And yes, I got a keyboard with both rows, they exist (inputs are sooooo fun, right?).

Reply:
```
$ grep -E 'XK_F[0-9]' /usr/include/X11/keysymdef.h 
#define XK_F1                            0xffbe
#define XK_F2                            0xffbf
#define XK_F3                            0xffc0
[snip]
#define XK_F34                           0xffdf
#define XK_F35                           0xffe0
```

Reply:
I decided to make this its own commit (1a423e8eb597db5157930ecfba2ed125d39a0036) and squash them later in a rebase. Should make it easier to review this.

_Thirty-five function keys_, that's crazy.